### PR TITLE
Allow attach for all Cosmos Account types

### LINKED ISSUE #227: Connect using ConnectionString
You need to have access to the subscription to be able to use the tool. I have multiple DocumentDB deployed on my clients where I have only access using the ConnectionString.
Please add a feature to add an Account out of a ConnectionString

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
I basically made the 'Attach mongo server' command generic. I also added 'Copy Connection String' to the non-mongo accounts.

I kept this feature in mind when I refactored for the shared tree provider, so it wasn't a ton of extra work.

Fixes #227

### CODE REVIEW

**Comment by EricJizbaMSFT** [comment]
Related to #188

**Comment by EricJizbaMSFT** [comment]
ping @StephenWeatherford and @PrashanthCorp now that 0.2.2 is behind us

File: src/tree/AttachedServersTreeItem.ts [hunk 1]
```diff
@@ -42,26 +51,40 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
...
-            } else {
-                this._attachedServers.push(node);
-                if (this._keytar) {
-                    await this._keytar.setPassword(this._serviceName, node.id, connectionString);
-                    await this.persistIds();
+    public async attachNewAccount(): Promise<void> {
+        const defaultExperience = <Experience>await vscode.window.showQuickPick(Object.keys(Experience), { placeHolder: "Select a Database Account API...", ignoreFocusOut: true });
+        if (defaultExperience) {
+            let placeholder: string = 'AccountEndpoint=...;AccountKey=...'
+            let validateInput = AttachedAccountsTreeItem.validateDocDBConnectionString;
```
Comment (PrashanthCorp) [inline src/tree/AttachedServersTreeItem.ts]: I feel placing lines 65,66 in the else block of the following if statement would be clearer. No strong opinions either way though.

Comment (EricJizbaMSFT) [reply src/tree/AttachedServersTreeItem.ts]: No strong preference either. Updated to reflect your suggestion

File: src/tree/AttachedServersTreeItem.ts [hunk 2]
```diff
@@ -42,26 +51,44 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
     }
 
     public async loadMoreChildren(_node: IAzureNode, clearCache: boolean): Promise<IAzureTreeItem[]> {
```
Comment (StephenWeatherford) [inline src/tree/AttachedServersTreeItem.ts]: Will it be obvious how to attach an account after the first one?

File: src/tree/AttachedServersTreeItem.ts [hunk 3]
```diff
@@ -105,17 +132,97 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
     private async loadPersistedServers() {
         const value: any = this._globalState.get(this._serviceName);
```
Comment (StephenWeatherford) [inline src/tree/AttachedServersTreeItem.ts]: Error handling?  I.e. settings are corrupt.  Or they have a later version of the extension then backgrade to this version, and the format has changed.

File: src/tree/AttachedServersTreeItem.ts [hunk 4]
```diff
@@ -105,17 +132,97 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
...
+
+            switch (api) {
+                case Experience.Table:
+                    treeItem = new TableAccountTreeItem(id, id, endpoint, masterKey);
+                    break;
+                case Experience.Graph:
+                    treeItem = new GraphAccountTreeItem(id, id, endpoint, masterKey);
+                    break;
+                case Experience.DocumentDB:
+                default:
```
Comment (StephenWeatherford) [inline src/tree/AttachedServersTreeItem.ts]: Wouldn't it be better to explicitly reference all handled cases and throw on default?

File: src/tree/AttachedServersTreeItem.ts [hunk 5]
```diff
@@ -105,17 +132,97 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
...
-        const value = this._attachedServers.map(node => node.id);
+        const value: IPersistedAccount[] = this._attachedAccounts.map((node: IAzureTreeItem) => {
+            let experience: Experience;
+            if (node instanceof MongoAccountTreeItem) {
+                experience = Experience.MongoDB;
+            } else if (node instanceof GraphAccountTreeItem) {
+                experience = Experience.Graph;
+            } else if (node instanceof TableAccountTreeItem) {
+                experience = Experience.Table;
+            } else {
```
Comment (StephenWeatherford) [inline src/tree/AttachedServersTreeItem.ts]: ditto

File: src/tree/AttachedServersTreeItem.ts [hunk 6]
```diff
@@ -42,26 +51,44 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
...
-            prompt: 'Enter the connection string for your database account',
-            ignoreFocusOut: true
-        });
-
-        if (connectionString) {
-            const id = await this.getServerIdFromConnectionString(connectionString);
-            const node = new MongoAccountTreeItem(id, id, connectionString, 'mongoServer');
-            if (this._attachedServers.find(s => s.id === node.id)) {
-                vscode.window.showWarningMessage(`Mongo server '${node.id}' is already attached.`)
+    public async attachNewAccount(): Promise<void> {
```
Comment (StephenWeatherford) [inline src/tree/AttachedServersTreeItem.ts]: Cool!  I can attach my graph accounts using the DocDB API and see vertices and edges in the treeview!

Comment (StephenWeatherford) [reply src/tree/AttachedServersTreeItem.ts]: But update didn't work. :-)

Comment (EricJizbaMSFT) [reply src/tree/AttachedServersTreeItem.ts]: Haha yes I suppose you can do that - what error did you get? I successfully updated just now...

**Comment by EricJizbaMSFT** [comment]
ping @StephenWeatherford

**Pull request merged**